File: Tank_Game/Assets/Universal Assets/ZDelete/Javascript/WorldBoundary.js
#pragma strict
#pragma implicit
#pragma downcast

// If worried about it going so fast it flys out of level then add an array that stores objects that fly away
// and if they don't enter into sphere again then delete them after a short pause;
function OnTriggerExit (other : Collider) {
	var basic : PlayerBasic = other.GetComponent(PlayerBasic);
	if (basic) {
		basic.SendMessage("ResetPosition");
	}
	else {
		Destroy(other.gameObject);
	}
}

File: Tank_Game/Assets/Universal Assets/WorldBoundary.cs
public class WorldBoundary : UnityEngine.MonoBehaviour {
	// If worried about it going so fast it flys out of level then add an array that stores objects that fly away
	// and if they don't enter into sphere again then delete them after a short pause;
	private void OnTriggerExit (UnityEngine.Collider other) {
		PlayerBasic basic = other.GetComponent(typeof(PlayerBasic)) as PlayerBasic;
		if (basic) {
			basic.ResetPosition();
		}
		else {
			UnityEngine.Object.Destroy(other.gameObject);
		}
	}
}
